### DISCUSSION TITLE
RadioLib building using Arduino IDE 2.0 doesn't respect EXCLUDES

### DISCUSSION THREAD

**Discussion by MrSniffer** [jgromes/RadioLib#591] [Q&A] [answered]
I modified my BuildOpt.h file to exclude a zillion protocols that I don't use (to speed up compilation and make sure no code and RAM space were being utilized.) But when I compile using Arduino IDE 2.0, all the excluded modules are rebuilt. How do I prevent this from happening?

Attached are my BuildOpt.h (as .txt) and a copy of the Windows commands executed by Arduino.
[RadioLib Build Files.txt](https://github.com/jgromes/RadioLib/files/9824765/RadioLib.Build.Files.txt)
[BuildOpt.h.txt](https://github.com/jgromes/RadioLib/files/9824783/BuildOpt.h.txt)

**Comment by jgromes** [comment, marked as answer]
Even if you enable the exclude macros, the files will remain in the build log. They are still physically present in the filesystem, and so they will be compiled (because the entire src directory is included). The difference is that the resulting object file will be empty or nearly empty. This speeds up compilation if the platform is poorly optimized, or if your PC is truly ancient. On most common platforms and PCs manufactured in the last decade, the difference should be negligible.

Regarding code and RAM space usage, you don't have to use the exclude macros because the compiler will automatically find everything that is not used and remove it from the final binary.

The primary use-case for exclusion macros is to resolve conflicts between RadioLib-defined and platform-defined classes or types.

**Reply by MrSniffer** [reply]
Thanks for the insights. As to the pain of re-compiling periodically, sounds like you haven't experienced the pain of Arduino IDE 2.0. Aaaaaaaaaaaaaaargh.
